lib/dogma/script.ex

```diff
@@ -118,6 +119,44 @@ defmodule Dogma.Script do
     errors
   end
 
+  @doc """
+  Builds the corrected_source by running the corrections
+  for any rule that has implemented correction/2.
+  Then replaces the file at at the path with the corrected source.
+  The correction is run once for each rule.
+  """
+  def repair(script, io \\ File) do
+    script.errors
+    |> Enum.group_by(&Map.get(&1, :rule))
+    |> Enum.reduce(script, &repair_violation/2)
+    |> write_corrected_source(io)
+  end
+
+  defp repair_violation({rule, violations}, script) do
+    if function_exported?(rule, :correction, 2) do
+      source = script.corrected_source || script.source
+      errors = Enum.map(script.errors, fn err ->
+        if Enum.member?(violations, err) do
+          %Error{ err | fixed?: true}
+        else
+          err
+        end
+      end)
```

Comment:
I think this function is hard to read. Perhaps repairing should have its own module, and this one should delegate to it.